insonmnia/oracle/oracle.go

```diff
@@ -101,12 +125,14 @@ func (o *Oracle) listenEventsRoutine(ctx context.Context) error {
 			}
 			switch value := event.Data.(type) {
 			case *blockchain.SubmissionData:
-				o.logger.Debug("new submission found", zap.Any("event", event))
-				if o.transactionValid(ctx, value.TransactionId) {
-					o.confirmChanging(ctx, value.TransactionId)
-				} else {
-					o.logger.Info("failed to confirm transaction, transaction invalid ")
+				o.logger.Debug("new submission found", zap.Any("event", *event))
+
+				if !o.transactionValid(ctx, value.TransactionId) {
+					o.logger.Info("failed to confirm transaction, transaction invalid",
```

Comment:
Seems like you're stepping into some unusual condition, maybe change log level to a warning?